lib/bolt/config.rb

```diff
@@ -32,12 +24,16 @@ def initialize(transport, uri = nil)
   end
 
   class Config
-    attr_accessor :concurrency, :format, :trace, :log, :puppetdb, :color, :save_rerun,
-                  :transport, :transports, :inventoryfile, :compile_concurrency, :boltdir,
-                  :puppetfile_config, :plugins, :plugin_hooks, :trusted_external,
-                  :apply_settings
-    attr_writer :modulepath
-    attr_reader :config_files, :warnings
+    attr_reader :config_files, :warnings, :data, :transports, :boltdir
+
+    TRANSPORT_CONFIG = {
+      'ssh'    => Bolt::Config::SSH,
+      'winrm'  => Bolt::Config::WinRM,
+      'pcp'    => Bolt::Config::Orch,
+      'local'  => Bolt::Util.windows? ? Bolt::Config::LocalWindows : Bolt::Config::Local,
```

Comment:
Would it be better for the differences in behavior to be implemented on a single `Bolt::Config::Local` class? Maybe the class ought to override `self.options` to provide dynamic behavior at runtime rather than branching at load time. That would allow us to test both versions of it (by stubbing `Bolt::Util.windows?`) on a single platform.

Reply:
Ah I see that part of the complication there is that `Local` derives from `Sudoable` whereas `LocalWindows` doesn't. With my other suggestion to remove the `Sudoable` config class, that issue goes away.

We might also want to consider a similar refactor of the actual local transport as well, though not as part of this PR. That may be informed by our plan to separate "transport" from "shell" though.

Reply:
Great idea :) Updated to use a single `Bolt::Config::Transport::Local` class.